### PR TITLE
Fix running ssh command on multiple nodes with mfa per session

### LINKED ISSUE #15497: tsh ssh user@key=val fails on multiple hosts with require_session_mfa
I enabled per session mfa for all hosts. I have two instances with `env: staging` label. I can ssh individually with tapping my key, but when running a command that matches multiple hosts it fails.

I expect `tsh` to ask me to tap the key for each host to execute a command in sequence. Instead it asks me to tap twice in a row, and then fails for all hosts:

```
tsh ssh user@env=staging ls /home
Tap any security key
WARNING: Multiple nodes matched label selector, running command on all.
Tap any security key
rpc error: code = PermissionDenied desc = MFA response validation failed: Error validating challenge
failed to authenticate using all MFA devices, rerun the command with '-d' to see error details for each device
Enter password for Teleport user klizhentas@gmail.com:
ERROR: failed reading prompt response
context canceled
```

Expected behavior:

```bash
tsh ssh user@env=staging ls
# tap to execute for each host separately
Tap any security key
... ls output on instance-1
Tap any security key 
... ls output on instance-2
```

Logs:

```
DEBU             FIDO2: Got 1 assertions webauthncli/fido2.go:185
DEBU             FIDO2: Authenticated: credential ID (b64) = syuNKHtFELz0I8rHztRRmw, user ID (hex) = some hex, user name = "" webauthncli/fido2.go:193
DEBU [CLIENT]    WEBAUTHN authentication failed error:[
ERROR REPORT:
Original Error: *errors.withStack failed to open: rx error
Stack Trace:
	/go/src/github.com/gravitational/teleport/lib/auth/webauthncli/fido2.go:643 github.com/gravitational/teleport/lib/auth/webauthncli.findSuitableDevices
	/go/src/github.com/gravitational/teleport/lib/auth/webauthncli/fido2.go:542 github.com/gravitational/teleport/lib/auth/webauthncli.runOnFIDO2Devices
	/go/src/github.com/gravitational/teleport/lib/auth/webauthncli/fido2.go:208 github.com/gravitational/teleport/lib/auth/webauthncli.fido2Login
	/go/src/github.com/gravitational/teleport/lib/auth/webauthncli/fido2_common.go:46 github.com/gravitational/teleport/lib/auth/webauthncli.FIDO2Login
	/go/src/github.com/gravitational/teleport/lib/auth/webauthncli/api.go:136 github.com/gravitational/teleport/lib/auth/webauthncli.crossPlatformLogin
	/go/src/github.com/gravitational/teleport/lib/auth/webauthncli/api.go:126 github.com/gravitational/teleport/lib/auth/webauthncli.Login
	/go/src/github.com/gravitational/teleport/lib/client/mfa.go:222 github.com/gravitational/teleport/lib/client.PromptMFAChallenge.func3
	/opt/go/src/runtime/asm_amd64.s:1571 runtime.goexit
User Message: device /dev/hidraw1: info
	failed to open: rx error] client/mfa.go:233
failed to authenticate using all MFA devices, rerun the command with '-d' to see error details for each device
rpc error: code = PermissionDenied desc = MFA response validation failed: Error validating challenge

ERROR REPORT:
Original Error: *status.Error rpc error: code = PermissionDenied desc = MFA response validation failed: Error validating challenge
Stack Trace:
	/go/src/github.com/gravitational/teleport/lib/client/client.go:494 github.com/gravitational/teleport/lib/client.(*ProxyClient).IssueUserCertsWithMFA
	/go/src/github.com/gravitational/teleport/lib/client/client.go:2139 github.com/gravitational/teleport/lib/client.(*ProxyClient).sessionSSHCertificate
	/go/src/github.com/gravitational/teleport/lib/client/client.go:1519 github.com/gravitational/teleport/lib/client.(*ProxyClient).ConnectToNode
	/go/src/github.com/gravitational/teleport/lib/client/api.go:2688 github.com/gravitational/teleport/lib/client.(*TeleportClient).runCommandOnNodes.func1
	/opt/go/src/runtime/asm_amd64.s:1571 runtime.goexit
User Message: rpc error: code = PermissionDenied desc = MFA response validation failed: Error validating challenge
```

```
tsh version
Teleport v10.1.2 git:v10.1.2-0-g5bc7caf go1.18.3
Proxy version: 10.1.2
```

```
teleport version
Teleport v10.1.2 git:v10.1.2-0-g5bc7caf go1.18.3
```

### PR DESCRIPTION
**Pull request by tobiaszheller**
When executing command on multiple nodes with MFA per session enabled following things happened:
1. Login to first node (with MFA)
2. Concurrent logins to all nodes (with MFA).
So if you should have 2 nodes based on labels, you will get 3 requests for MFA, additionally those last 2 will be run concurrently which results in errors (probably some race conditions) and weird UX. 

It was solved by running commands sequentially when MFA per session is enabled.

Note that right now we we returning first error instead of last when any of commands fails (i guess this is fine).

I have also need to move `PromptWebauthn` from `export_test.go` to `export.go` so it can be accessible from test in other packages (`tool/tsh` in that case).

Fixes: https://github.com/gravitational/teleport/issues/15497

### CODE REVIEW

File: lib/client/api.go [hunk 1]
```diff
@@ -2715,37 +2733,55 @@ func (tc *TeleportClient) ListKubernetesClustersWithFiltersAllClusters(ctx conte
...
+	defer clt.Close()
+
+	// Let's check is first node has mfa required.
+	// If it's required, run commands sequentially to avoid
+	// race conditions and weird ux during mfa.
+	mfaRequiredCheck, err := clt.IsMFARequired(ctx, &proto.IsMFARequiredRequest{
+		Target: &proto.IsMFARequiredRequest_Node{
+			Node: &proto.NodeLogin{
+				// Let's check only single node, all must have the same settings.
+				// TODO(rosstimothy): is it right?
```
Comment (tobiaszheller) [inline lib/client/api.go:2749]: @rosstimothy can you confirm it?

Comment (tobiaszheller) [reply lib/client/api.go:2749]: @rosstimothy maybe I am missing something but I don't think that  this check has been performed before. 
Connecting to node is happening after I am checking if this is required: https://github.com/gravitational/teleport/blob/408ac68f934d11014c93571270a695fd2e6f6dd1/lib/client/api.go#L2745-L2767
Moreover this check happens only if this ssh command should be executed on multiple ssh nodes, so I guess user can accept that additional latency.

Comment (tobiaszheller) [reply lib/client/api.go]: @rosstimothy I have added estabilishing connection to auth before as separate commit, but to be honest I don't like. ConnectToNode is used in a lot of places and we can either:
1. always dial auth and pass it to every `ConnectToNode`
2. or pass client as optional (i don't like passing client as optional).
3. Add new function `ConnectToNodeWithAuthClient` and do the same with ` proxy.IssueUserCertsWithMFA` and `sessionSSHCertificate`. This seems also like suboptimal solution.

So I have decided go with 1. But I am afraid that it may break something, because now we are doing that dial much earlier. UT are passing however.
One place where it can fail is https://github.com/gravitational/teleport/blob/0532344905800b6fdf3d497509478df5b70dbfff/lib/client/client.go#L2143-L2150. It was always executed before dialing to cluster auth. 🤔 

I ahve also tried passing only info about required check but it is also very ugly, a lot of ifs in `proxy.IssueUserCertsWithMFA` and we need to pass that parameter or optional in a lot places.

So to sum up, i would prefer that additional latency only on running ssh on multiple host. But if you still think we should go with passing auth client, please review my newest commit.

Comment (tobiaszheller) [reply lib/client/api.go]: > What are your thoughts about renaming NodeAddr to NodeDetails and adding a field to that struct and to ReissueParams which indicates if mfa is required?

Here we have 2 paths:
1. Always require that `is_mfa_required` which will result in changing it in a lot places. 
2. Allow passing it as optional parameter - but it means that is needs to be `required|notRequired|unknown` and it makes `proxy.IssueUserCertsWithMFA` not readable.

Solution 1 here is very similar that what I did with calling auth earlier (because that auth will be necessarny anyway to check `is_mfa_required`. So after all I think that solution which is implemented now (with passing auth client) is better. 
What do you think?

Comment (tobiaszheller) [reply lib/client/api.go]: @rosstimothy should be ready to re-review

**Review by rosstimothy — APPROVED** [review]

**Comment by tobiaszheller** [comment]
@kimlisa @probakowski it should be ready for review now

**Review by kimlisa — APPROVED** [review]
